sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/Column.java

```diff
@@ -349,7 +349,7 @@ public void addValue(Type type, Object field) {
         break;
       case FLOAT_TYPE:
         nulls.set(size, field == null);
-        doubleVars()[size] = field == null ? 0 : ((Float)field).doubleValue();
+        doubleVars()[size] = field == null ? 0 : new Double(field.toString());
```

Comment:
if the problem is the precision, isn't enough to cast it to Double instead of creating a double out of a string?

Reply:
sorry i am not sure what is your meaning.   "(Double)field" ?

Reply:
I mean `((Double)field)`

Reply:
oh it is wrong. we cannot cast Float to Double like this.

Reply:
I see now the issue, so we cannot convert directly from float to double, we have to pass through another object which may be a string as in this your solution. Please disregard my comment, thanks.